types/mongoose/index.d.ts

```diff
@@ -2453,10 +2419,10 @@ declare module "mongoose" {
      * Performs aggregations on the models collection.
      * If a callback is passed, the aggregate is executed and a Promise is returned.
      * If a callback is not passed, the aggregate itself is returned.
-     * @param ... aggregation pipeline operator(s) or operator array
+     * @param aggregations pipeline operator(s) or operator array
      */
-    aggregate(...aggregations: Object[]): Aggregate<Object[]>;
-    aggregate(...aggregationsWithCallback: Object[]): Promise<Object[]>;
+    aggregate(aggregations: Object[]): Aggregate<Object[]>;
```

Comment:
This can still take the aggregations as arguments instead of a single array. So we'll still have to use `...aggregations`.
See: http://mongoosejs.com/docs/api.html#aggregate_aggregate

They give the example:
```js
Users.aggregate(
  { $group: { _id: null, maxBalance: { $max: '$balance' }}},
  { $project: { _id: 0, maxBalance: 1 }},
  function (err, res) {
    if (err) return handleError(err);
    console.log(res); // [ { maxBalance: 98000 } ]
  });
```
^Could you add this as a test case and make sure it passes? I forgot to do this from a while back.